framework_tvb/tvb/core/adapters/abcadapter.py

```diff
@@ -550,6 +569,47 @@ def load_view_model(self, operation):
         input_gid = json.loads(operation.parameters)['gid']
         return h5.load_view_model(input_gid, storage_path)
 
+    @staticmethod
+    def gather_all_references_by_index(h5_file, ref_files):
+        refs = h5_file.gather_references()
+        for _, gid in refs:
+            if not gid:
+                continue
+            index = ABCAdapter.load_entity_by_gid(gid)
+            h5_file = h5.h5_file_for_index(index)
+            ref_files.append(h5_file.path)
+            ABCAdapter.gather_all_references_by_index(h5_file, ref_files)
+
+    @staticmethod
+    def gather_all_references_of_view_model(gid, base_dir, ref_files):
+        vm_path = h5.determine_filepath(gid, base_dir)
+        ref_files.append(vm_path)
+        view_model_class = H5File.determine_type(vm_path)
+        view_model = view_model_class()
+
+        with ViewModelH5(vm_path, view_model) as vm_h5:
+            references = vm_h5.gather_references()
+            uuids = vm_h5.gather_references_by_uuid()
+
+            for _, gid in references:
+                if not gid:
+                    continue
+                if isinstance(gid, (list, tuple)):
+                    for list_gid in gid:
+                        ABCAdapter.gather_all_references_of_view_model(list_gid, base_dir, ref_files)
+                else:
+                    ABCAdapter.gather_all_references_of_view_model(gid, base_dir, ref_files)
+
+            uuid_files = []
+            for _, gid in uuids:
+                if not gid:
+                    continue
+                index = ABCAdapter.load_entity_by_gid(gid)
+                h5_file = h5.h5_file_for_index(index)
+                uuid_files.append(h5_file.path)
+                ABCAdapter.gather_all_references_by_index(h5_file, uuid_files)
+            ref_files.extend(uuid_files)
+
```

Comment:
As we already have (and it seems to be better) `h5.load_view_model` could we also maybe move there `gather_all_references_by_index ` and `gather_all_references_of_view_model `  ?
Or next to `load_entity_by_gid` in tvb.core.entities.load.py ?